examples/angular/src/app/resolver/dotcms-page.resolver.ts

```diff
@@ -48,8 +49,24 @@ export const DotCMSPageResolver = async (
     navRequest,
   ]);
 
-  const pageAsset = pageResponse.entity;
+  let pageAsset = pageResponse.entity;
   const nav = navResponse.entity;
 
+  const { vanityUrl } = pageAsset;
+
+  if (vanityUrl?.permanentRedirect) {
+    router.navigate([vanityUrl.forwardTo]);
```

Comment:
@KevinDavilaDotCMS Should this redirect in the canMatchFn take advantage in this guar also make a request to obtain this data?

Reply:
Doing so will match what @rjvelazco did.

Reply:
And actually if the data is already in the router, we don't need a provider.

Reply:
Moved!